pkg/body-based-routing/README.md

```diff
@@ -0,0 +1,9 @@
+# Body-Based Routing
+This package provides an extension that can be deployed to write the `model`
+HTTP body parameter as a header so as to enable routing capabilities on the
+model name.
+
+As per OpenAI spec, it is standard for the model name to be included in the
+body of the HTTP request. However, most implementations do not support routing
+based on the request body. This extension helps bridge the gap for clients that
+are unable to write the model name into the headers themselves.
```

Comment:
Probably worth adding some additional text describing how this can be integrated:

```suggestion
This extension works by parsing the request body. If it finds a `model` parameter in the
request body, it will copy the value of that parameter into a request header.

This extension is intended to be paired with an `ext_proc` capable Gateway. There is not
a standard way to represent this kind of extension in Gateway API yet, so we recommend
referring to implementation-specific documentation for how to deploy this extension.
```